openverse_api/catalog/api/models/image.py

```diff
@@ -6,14 +6,31 @@
     AbstractMediaList,
     AbstractMediaReport,
 )
+from catalog.api.models.mixins import FileMixin
 from django.db import models
 from uuslug import uuslug
 
 
-class Image(AbstractMedia):
+class ImageFileMixin(FileMixin):
+    """
+    This mixin adds fields related to image resolution to the standard file
+    mixin. Do not use this as the sole base class.
+    """
+
     width = models.IntegerField(blank=True, null=True)
     height = models.IntegerField(blank=True, null=True)
 
+    @property
+    def resolution_in_mp(self):  # ~ MP or megapixels
+        return (self.width * self.height) / 1e6
+
+    class Meta:
+        abstract = True
+
+
+class Image(ImageFileMixin, AbstractMedia):
+    pass
```

Comment:
Neat!

Reply:
Once we abstract the file-specific outside the `Image` model, it becomes empty! Maybe we can add `alt_files` for images in the future.